### ISSUE TITLE
Pymatgen import error with OFM

### ISSUE THREAD

**Issue by arosen93** [arosen93/QMOF#17]
Moved from https://github.com/arosen93/QMOF/discussions/16, which stated:
_This is my prolem,ImportError: cannot import name 'VoronoiCoordFinder' from 'pymatgen.analysis.structure_analyzer'.
What is the version of pymatgen package?Has the function been removed? How can I solve this problem?
Thank you very much._

Hi @tuantuan-lin. I'm happy to try and help. Pymatgen recently introduced some backwards-incompatible changes with how functions are called, as noted [here](https://pymatgen.org/#major-announcement-v2022-0). I imagine this is the likely cause.

In order to help you, let's make sure we're talking about the same thing. I assume that you are referring to [`ofm_feature_generator.py`](https://github.com/arosen93/QMOF/blob/main/machine_learning/orbital_field_matrix/ofm_feature_generator.py), correct? And the problem appears upon calling the `OrbitalFieldMatrix` function from matminer? Please let me know the full traceback error (e.g. what lines it's failing on).

If this is the case, please try the following:
1. Uninstall matminer.
2. Uninstall pymatgen.
3. Install the most recent version of pymatgen (`pip install pymatgen`).
4. Clone the current matminer repository (`git clone https://github.com/hackingmaterials/matminer.git`).
5. In the base directory of the downloaded matminer repository, install matminer via `pip install .`
6. Try the code again. 

As for the specific versions used in the QMOF Database paper, please see the "Additional Software and Hardware Details for Machine Learning" section (bottom of pg 8) in the [Supporting Information](https://ars.els-cdn.com/content/image/1-s2.0-S2590238521000709-mmc1.pdf). I used Pymatgen v.2020.12.3 with matminer v.0.6.4.

**Comment by tuantuan-lin** [comment]
Thank you very much for your help. The problem does appear upon calling the OrbitalFieldMatrix function from matminer.And I tried the methods you gave, there are still the following problems:
`Traceback (most recent call last):
  File "C:\Users\Administrator.DESKTOP-M4OGFT6\AppData\Roaming\Python\Python37\site-packages\IPython\core\interactiveshell.py", line 3441, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-76dad9573da9>", line 1, in <module>
    runfile('E:/1/10-Model/04-OFM/ofm_feature_generator.py', wdir='E:/1/10-Model/04-OFM')
  File "D:\Program Files\JetBrains\PyCharm 2019.3.3\plugins\python\helpers\pydev\_pydev_bundle\pydev_umd.py", line 197, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "D:\Program Files\JetBrains\PyCharm 2019.3.3\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "E:/1/10-Model/04-OFM/ofm_feature_generator.py", line 1, in <module>
    from matminer.featurizers.structure import OrbitalFieldMatrix
  File "D:\Program Files\JetBrains\PyCharm 2019.3.3\plugins\python\helpers\pydev\_pydev_bundle\pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "D:\ProgramData\Anaconda3\lib\site-packages\matminer\featurizers\structure.py", line 21, in <module>
    from pymatgen import Structure, Lattice
ImportError: cannot import name 'Structure' from 'pymatgen' (unknown location)
`
So I changed `from pymatgen import Structure, Lattice` to `from pymatgen.core import Structure, Lattice` in file structure.py.But there is still the following problem:
`Traceback (most recent call last):
  File "C:\Users\Administrator.DESKTOP-M4OGFT6\AppData\Roaming\Python\Python37\site-packages\IPython\core\interactiveshell.py", line 3441, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-76dad9573da9>", line 1, in <module>
    runfile('E:/1/10-Model/04-OFM/ofm_feature_generator.py', wdir='E:/1/10-Model/04-OFM')
  File "D:\Program Files\JetBrains\PyCharm 2019.3.3\plugins\python\helpers\pydev\_pydev_bundle\pydev_umd.py", line 197, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "D:\Program Files\JetBrains\PyCharm 2019.3.3\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "E:/1/10-Model/04-OFM/ofm_feature_generator.py", line 1, in <module>
    from matminer.featurizers.structure import OrbitalFieldMatrix
  File "D:\Program Files\JetBrains\PyCharm 2019.3.3\plugins\python\helpers\pydev\_pydev_bundle\pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "D:\ProgramData\Anaconda3\lib\site-packages\matminer\featurizers\structure.py", line 34, in <module>
    from matminer.featurizers.site import OPSiteFingerprint, \
  File "D:\Program Files\JetBrains\PyCharm 2019.3.3\plugins\python\helpers\pydev\_pydev_bundle\pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "D:\ProgramData\Anaconda3\lib\site-packages\matminer\featurizers\site.py", line 11, in <module>
    from matminer.utils.data import MagpieData
  File "D:\Program Files\JetBrains\PyCharm 2019.3.3\plugins\python\helpers\pydev\_pydev_bundle\pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "D:\ProgramData\Anaconda3\lib\site-packages\matminer\utils\data.py", line 17, in <module>
    from pymatgen import Element
ImportError: cannot import name 'Element' from 'pymatgen' (unknown location)
`
What can I do to correct my mistakes?
Thank you very much !

**Comment by arosen93** [comment]
From your information, it looks like there is a conflict between Pymatgen and matminer, as I alluded to.

You have two options:
1. Use a version of Pymatgen that is before the backwards-incompatible changes introduced in v.2022.0.*.
2. Use the most recent version of matminer on GitHub, which has been updated for Pymatgen v.2022.0* and newer.

In my prior message, I walked through the steps on how to address Option 2. However, it looks like you did not properly uninstall your version of matminer to replace it with the most recent version on GitHub. This is most clearly seen in the following error message:

```
File "D:\ProgramData\Anaconda3\lib\site-packages\matminer\featurizers\structure.py", line 21, in from pymatgen import Structure, Lattice
ImportError: cannot import name 'Structure' from 'pymatgen' (unknown location)
```
The [current version of matminer on GitHub](https://github.com/hackingmaterials/matminer/blob/46d6a90664dc9e804e81c2c22cbee9e7221e8315/matminer/featurizers/structure.py#L21) already has the `from pymatgen.core import Structure, Lattice`, which means that you are not using this version.
